packages/mobile/src/fees/saga.test.ts

```diff
@@ -1,79 +1,230 @@
+          estimation: estimation(new BigNumber(0.01).toString()),
+        })
       )
       .run()
   })
 
-  it('updates the default send fee', async () => {
-    await expectSaga(estimateFeeSaga, { feeType: FeeType.SEND })
+  it('estimates the dek register fee', async () => {
+    const kit = await getContractKitAsync()
+    const mockAccountsWrapper = { setAccount: jest.fn(() => ({ txo: mockTxo })) }
+    await expectSaga(estimateFeeSaga, {
+      payload: { feeType: FeeType.REGISTER_DEK, tokenAddress: mockCusdAddress },
+    })
+      .withState(createMockStore({}).getState())
       .provide([
-        [call(getConnectedAccount), mockAccount],
-        [matchers.call.fn(getSendTxGas), new BigNumber(GAS_AMOUNT)],
-        [select(cUsdBalanceSelector), '1'],
+        [call(getContractKit), kit],
+        [call([kit.contracts, kit.contracts.getAccounts]), mockAccountsWrapper],
+        [matchers.call.fn(estimateGas), new BigNumber(GAS_AMOUNT)],
+        [
+          call(calculateFee, new BigNumber(GAS_AMOUNT), Currency.Dollar),
+          { fee: new BigNumber(1e16), currency: Currency.Dollar },
+        ],
       ])
       .put(
-        feeEstimated(
-          FeeType.SEND,
-          new BigNumber(10000).times(GAS_PRICE_INFLATION_FACTOR).times(GAS_AMOUNT).toString()
-        )
+        feeEstimated({
+          feeType: FeeType.REGISTER_DEK,
+          tokenAddress: mockCusdAddress,
+          estimation: estimation(new BigNumber(0.01).toString()),
+        })
       )
       .run()
   })
 
-  it('updates the default escrow reclaim fee', async () => {
-    await expectSaga(estimateFeeSaga, { feeType: FeeType.SEND })
+  it('marks as error if no paymentID is sent for escrow reclaim fee', async () => {
+    await expectSaga(estimateFeeSaga, {
+      payload: { feeType: FeeType.RECLAIM_ESCROW, tokenAddress: mockCusdAddress },
+    })
+      .withState(createMockStore({}).getState())
       .provide([
-        [call(getConnectedAccount), mockAccount],
-        [matchers.call.fn(getReclaimEscrowGas), new BigNumber(GAS_AMOUNT)],
-        [select(cUsdBalanceSelector), '1'],
+        [call(createReclaimTransaction, 'paymentID'), mockTxo],
+        [matchers.call.fn(estimateGas), new BigNumber(GAS_AMOUNT)],
+        [
+          call(calculateFee, new BigNumber(GAS_AMOUNT), Currency.Dollar),
+          { fee: new BigNumber(1e16), currency: Currency.Dollar },
+        ],
       ])
       .put(
-        feeEstimated(
-          FeeType.SEND,
-          new BigNumber(10000).times(GAS_PRICE_INFLATION_FACTOR).times(GAS_AMOUNT).toString()
-        )
+        feeEstimated({
+          feeType: FeeType.RECLAIM_ESCROW,
+          tokenAddress: mockCusdAddress,
+          estimation: {
+            loading: false,
+            error: true,
+            usdFee: null,
+            lastUpdated: Date.now(),
+          },
+        })
       )
       .run()
   })
 
-  it("doesn't calculates fee if the balance is zero", async () => {
```

Comment:
How does it handle the 0 balance case?

Reply:
It throws an error. If the balance is exactly zero there is a check that will fail and mark the estimation with `error` and if it's slightly more than zero but less than the required fee the fee estimation fails